src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,62 @@
 package mate.academy.lib;
 
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.util.HashMap;
+import java.util.Map;
+import mate.academy.service.FileReaderService;
+import mate.academy.service.ProductParser;
+import mate.academy.service.ProductService;
+
 public class Injector {
     private static final Injector injector = new Injector();
+    private final Map<Class<?>, Object> instances = new HashMap<>();
 
     public static Injector getInjector() {
         return injector;
     }
 
     public Object getInstance(Class<?> interfaceClazz) {
-        return null;
+        if (instances.containsKey(interfaceClazz)) {
+            return instances.get(interfaceClazz);
+        }
+        Class<?> clazz = findImplementation(interfaceClazz);
+        if (clazz == null || !clazz.isAnnotationPresent(Component.class)) {
```

Comment:
Ensure that the `Component` annotation is correctly applied to all classes that should be managed by the injector. If a class is missing this annotation, it will cause a `RuntimeException`.